elegant/clean_timepoint_data.py

```diff
@@ -0,0 +1,147 @@
+import pathlib
+import shutil
+import json
+import datetime
+
+import numpy
+
+from zplib import datafile
+from elegant import load_data
+
+def remove_timepoint_for_position(experiment_root, position, timepoint, dry_run=False):
+    """Removes all memory of a given timepoint for a given position (files, metadata, annotations entries)
+
+    Parameters
+        experiment_root - str/pathlib.Path pointing to experiment directory
+        position - str for position to modify
+        timepoint - str with standard format for timepoint acquisition (YYYY-MM-DDtHHMM)
+        dry_run - bool flag that toggles taking action (if False, only specifies when offending files are found)
+
+    Example Usage
+
+        experiment_root = /path/to/experiment
+        position = '000'
+        timepoint = '2018-07-30t1200'
+        dry_run = True  # No deleting; just see what happens
+        remove_timepoint_for_position(experiment_root, position, timepoint, dry_run=dry_run)
+    """
+
+    experiment_root = pathlib.Path(experiment_root)
+
+    files_to_remove = [img_file for img_file in (experiment_root / position).iterdir() if timepoint in str(img_file.name)]
```

Comment:
With string ops always choose the least-powerful/most-specific way of matching. Here do `if img_file.name.startswith(timepoint)`, rather than looking for the timepoint anywhere. (Also, `img_file.name` is a string already.)